src/hw_isolation_event/hw_status_manager.cpp

```diff
@@ -106,6 +121,183 @@ std::pair<event::EventMsg, event::EventSeverity>
+                    }
+
+                    if (hwasState.present)
+                    {
+                        ATTR_PHYS_BIN_PATH_Type physBinPath;
+                        if (DT_GET_PROP(ATTR_PHYS_BIN_PATH, tgt, physBinPath))
+                        {
+                            log<level::ERR>(
+                                fmt::format(
+                                    "Skipping to create the hardware "
+                                    "status event because failed to get "
+                                    "ATTR_PHYS_BIN_PATH from [{}]",
+                                    pdbg_target_path(tgt))
+                                    .c_str());
+                            continue;
+                        }
+
+                        devtree::DevTreePhysPath devTreePhysPath;
+                        std::copy(std::begin(physBinPath),
+                                  std::end(physBinPath),
+                                  std::back_inserter(devTreePhysPath));
+
+                        auto hwInventoryPath =
+                            _isolatableHWs.getInventoryPath(devTreePhysPath);
+
+                        if (!hwInventoryPath.has_value())
+                        {
+                            log<level::ERR>(
+                                fmt::format(
+                                    "Skipping to create the hardware "
+                                    "status event because unable to find "
+                                    "the inventory path for the given "
+                                    "hardware [{}]",
+                                    pdbg_target_path(tgt))
+                                    .c_str());
+                            continue;
+                        }
+
+                        event::EventMsg eventMsg;
+                        event::EventSeverity eventSeverity;
+                        record::entry::EntryErrLogPath eventErrLogPath;
+
+                        auto isolatedhwRecordInfo =
+                            _hwIsolationRecordMgr.getIsolatedHwRecordInfo(
+                                *hwInventoryPath);
+
+                        if (isolatedhwRecordInfo.has_value())
+                        {
+                            if (hwasState.functional)
+                            {
+                                /**
+                                 * Event is not required since the hardware
+                                 * isolation record is exist and not applied
+                                 * so far.
+                                 */
+                                continue;
+                            }
+
+                            // Error log might be present or not in the record.
+                            eventErrLogPath =
+                                (std::get<1>(*isolatedhwRecordInfo).empty()
+                                     ? ""
+                                     : std::get<1>(*isolatedhwRecordInfo));
```

Comment:
why complicate, i see this code is same as below one

eventErrLogPath =std::get<1>(*isolatedhwRecordInfo));

Reply:
Ack, Good catch (You found the one which I thought to change since `getIsolatedHwRecordInfo` returning as empty in the tuple if doesn't have the error log but, I forget to correct it.)